src/datepicker/datepicker.js

```diff
@@ -40,7 +40,7 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
   // Watchable date attributes
   angular.forEach(['minDate', 'maxDate'], function(key) {
     if ($attrs[key]) {
-      $scope.$parent.$watch($parse($attrs[key]), function(value) {
```

Comment:
Hmm, I guess this is technically ok, but by passing in $parse, I think it actually saves slightly on performance since $parse already has cached the string parsing of the attribute value.